### ISSUE TITLE
ShadowSQLiteConnection does not properly handle blocked insertions

### ISSUE THREAD

**Issue by quibi-jlk** [robolectric/robolectric#5267]
### Description

See https://issuetracker.google.com/issues/141625537 for full context. Essentially, insertions with ignore or abort conflict types should return -1, but instead they return whatever the previous inserted row ID was.

### Steps to Reproduce

This is super trivial to reproduce in a robolectric test with Room. Any dao function that does something like:

```
@Insert(onConflict = OnConflictStrategy.IGNORE)
abstract fun insertThing(thing: Thing): Long
```

is supposed to return -1 on conflict as per the Room docs. This works on real devices and fails in robolectric. I've tracked the issue down to this line:

https://github.com/robolectric/robolectric/blob/master/shadows/framework/src/main/java/org/robolectric/shadows/ShadowSQLiteConnection.java#L679

Which is not doing the same thing as native sqlite. Instead of returning -1 on conflict, it returns whatever the last insertion was. This makes it really hard to test some Room behavior.

### Robolectric & Android Version
Robolectric 4.3, Android Q

**Comment by CharlyLafon37** [comment]
I'm also having this issue and I can't find a workaround it. @quibi-jlk Did you manage to fix your tests without changing the DAO ?

**Comment by quibi-jlk** [comment]
Sorry @CharlyLafon37, it's been a while and I don't recall exactly what our workaround ended up being for this. I can say that now we've been using the `Room.inMemoryDatabaseBuilder` everywhere we can or we mock things out at a higher level to avoid this. It was definitely a super frustrating thing to track down, though.

**Comment by CharlyLafon37** [comment]
Yeah I understand very well the kind of frustration you've been through, I just spent 4 hours to track this bug down I was going insane. The only workaround for it I've found is to do in my DAO :

```
@Transaction
protected open suspend fun upsert(entity: T)
{
    insertIgnoreConflict(entity)
    update(entity)
}
```

instead of 

```
@Transaction
protected open suspend fun upsert(entity: T)
{
    insertIgnoreConflict(entity).also { if(it == -1L) update(entity) }
}
```

but I won't do that, it's just way less efficient.

Thanks for your response though, I think I'll create another issue to bring this bug all the way up the issues list, it's my only hope I fear 😄

**Comment by quibi-jlk** [comment]
Hah, yeah, I feel your pain! Good luck!

fyi @jongerrish
